File: src/my_project_typescript/easy_problems_typescript/from151to200/powXN.ts
function myPow(x: number, n: number): number {

    if (!n){
        return 1
    }
    if (n < 0){
        return 1/myPow(x,-n)
    }
    if (n%2 !== 0){
        return x * myPow(x,n-1)
    }
    return myPow(x*x,n/2)

};

File: src/my_project/interviews/top_150_questions_round_7/pow_x_n.py
from typing import List, Union, Collection, Mapping, Optional
from abc import ABC, abstractmethod

class Solution:
    def myPow(self, x:float, n:int):

        if not n: 
            return 1
        if n < 1:
            return self.myPow(1/x, -n)
        if n % 2: 
            return x*self.myPow(x, n-1)
        else: 
            return self.myPow(x**2, n//2)